service/javadsl/api/src/test/scala/com/lightbend/lagom/internal/api/ServiceReaderSpec.scala

```diff
@@ -38,6 +37,17 @@ class ServiceReaderSpec extends WordSpec with Matchers with Inside {
       endpoint.responseSerializer should ===(MessageSerializers.STRING)
     }
 
+    "fail to read a Java service descriptor from a public interface because the path parameter could not be serialized" in {
+      val caught = intercept[IllegalArgumentException] {
+        val descriptor = serviceDescriptor[InvalidPathParameterService]
+      }
+      caught.getMessage should ===("Error encountered while resolving the interface com.lightbend.lagom.api.mock.InvalidPathParameterService.helloservice" +
+        " call: No path parameter serializer was found for the class com.lightbend.lagom.api.mock.Data path parameter. This can be fixed either by implementing " +
+        "and then explicitly registering a com.lightbend.lagom.javadsl.api.PathParamSerializer for class com.lightbend.lagom.api.mock.Data on the interface " +
+        "com.lightbend.lagom.api.mock.InvalidPathParameterService service descriptor, or perhaps this parameter is meant to be the request message declared in the " +
+        "ServiceCall, and not extracted out of the path?")
+    }
+
```

Comment:
@silver-soule sorry, but I haven't seen that before. My bad. 

I find it fragile to test matching the text. We need to copy and paste and hack it to replace `method.getDeclaringClass + "." + method.getName`.

Maybe a better approach would be to create a new exception type that extends `IllegalArgumentException`. Later, when testing, we can simply check if we are getting the exception of the right type and we don't need to test the messaging.

Reply:
@renatocaval 
Yeah, that would be a better approach indeed , but I am not sure about two things -
1. Where should I create this new exception class? In https://github.com/lagom/lagom/tree/master/service/javadsl/api/src/main/scala/com/lightbend/lagom/internal/javadsl/api does not seem ideal to me.
2. Will calling this new subclass of IllegalArgumentException something like IllegalArgumentExceptionWrapper be okay?

Reply:
> Will calling this new subclass of IllegalArgumentException something like IllegalArgumentWrapperException be okay?

I think that name is not adding any information. `IllegalArgumentWrapperException` is a name that every possible java project could have. I think something like `MissingPathParamSerializerException` fits better the needs. WDYT?

Reply:
We can call it `IllegalPathParameterException`. 

We can add it under:
https://github.com/lagom/lagom/tree/master/service/javadsl/api/src/main/scala/com/lightbend/lagom/javadsl/api

Thus, not the `internal` package

Reply:
@ignasi35 @renatocaval Thanks for your  advice. I am going with IllegalPathParameterException, because it seems to align slightly better with the error message than MissingPathParamSerializerException.